cmake-modules/gcc-arm-toolchain.cmake

```diff
@@ -0,0 +1,5 @@
+set(CMAKE_SYSTEM_NAME Generic)
```

Comment:
Why `Generic`? What does that mean?

Reply:
@barcharcraz helped create this module and can more accurately describe what's going on here. I know that this is setting us up for cross-compilation and for a target system that is different from the host system https://cmake.org/cmake/help/v3.10/variable/CMAKE_SYSTEM_NAME.html